### PR TITLE
Fixes + docs

### PR DESCRIPTION
**Pull request by saurabhagas**

### CODE REVIEW

**Comment by saurabhagas** [comment]
@aurbroszniowski, @mobasherul PR ready.

File: client-internal/src/main/java/org/terracotta/angela/client/Tsa.java [hunk 1]
```diff
@@ -751,45 +747,46 @@ public ConfigToolExecutionResult detachNode(int stripeIndex, int serverIndex) {
...
           command.add("-s");
           command.add(stripe.get(i).getHostPort());
-        }
-        ConfigToolExecutionResult result = configTool(stripe.get(0)).executeCommand(command.toArray(new String[0]));
-        if (result.getExitStatus() != 0) {
-          throw new RuntimeException("ConfigTool::executeCommand with command parameters failed with: " + result);
+
+          ConfigToolExecutionResult result = configTool(stripe.get(0)).executeCommand(command.toArray(new String[0]));
+          if (result.getExitStatus() != 0) {
+            throw new RuntimeException("ConfigTool::executeCommand with command parameters failed with: " + result);
```
Comment (aurbroszniowski) [inline client-internal/src/main/java/org/terracotta/angela/client/Tsa.java:765]: there's a double ::

File: docs/src/main/java/GettingStarted.java [hunk 2]
```diff
@@ -37,23 +37,26 @@
...
   @Test
   public void configureCluster() throws Exception {
     // tag::configureCluster[]
     ConfigurationContext configContext = customConfigurationContext() // <1>
         .tsa(tsa -> tsa // <2>
             .topology(new Topology( // <3>
-                distribution(version(EHCACHE_OS_VERSION), PackageType.KIT, LicenseType.TERRACOTTA_OS), // <4>
-                tcConfig(version(EHCACHE_OS_VERSION), getClass().getResource("/tc-config-a.xml")))) // <5>
+                distribution(version("3.8.1"), PackageType.KIT, LicenseType.TERRACOTTA_OS), // <4>
+                tcConfig(version("3.8.1"), getClass().getResource("/tc-config-a.xml")))) // <5>
```
Comment (aurbroszniowski) [inline docs/src/main/java/GettingStarted.java:59]: why replacing this? I think that the reason to use EHCACHE_OS_VERSION was not to have to update test for every terracotta release

File: docs/src/main/java/GettingStarted.java [hunk 3]
```diff
@@ -97,13 +100,66 @@ public void showTsaApi() throws Exception {
...
   }
 
+  @Test
+  public void showDynamicTsaApi() throws Exception {
+    // tag::showDynamicTsaApi[]
+    ConfigurationContext configContext = customConfigurationContext()
+        .tsa(tsa -> tsa
+            .topology(
+                new Topology(
+                    distribution(version("3.9-SNAPSHOT"), KIT, TERRACOTTA_OS),
```
Comment (aurbroszniowski) [inline docs/src/main/java/GettingStarted.java:110]: there's an inconsistency regarding the version, above it is 3.8.1 and here 3.9-SNAPSHOT

File: docs/src/main/asciidoc/getting-started.adoc [hunk 4]
```diff
@@ -61,6 +61,20 @@ include::../java/GettingStarted.java[tags=showTsaApi]
 <9> Stop one Terracotta server
 <10> Get the current state of the Terracotta server
 
+== Dynamic Tsa API
+
+[source,java]
+----
+include::../java/GettingStarted.java[tags=showDynamicTsaApi]
+----
+<1> Define an intial dynamic cluster topology
```
Comment (aurbroszniowski) [inline docs/src/main/asciidoc/getting-started.adoc:70]: since this doc will be read by people migrating from fixed to dynamic config, it could be helpful to add a sentence to explain the different actions needed for a dynamic config:

something like
"A dynamic config doesn't rely on a tc-config.xml file, but is defined dynamically, therefore the Angela API provides a DynamicConfigManager.dynamicCluster() method to define within the test the different parameters.
It is possible to dynamically attach or detach stripes, which is done with the Tsa.attach() and Tsa.detach() API.
The config tool will activate attached Terracotta servers, which is done by the Tsa.activate() method"

feel free to rephrase

File: client-internal/src/main/java/org/terracotta/angela/client/Tsa.java (hunk 1 above)
Comment (saurabhagas) [reply client-internal/src/main/java/org/terracotta/angela/client/Tsa.java:765]: Yes, that's the scope resolution operator (the kind Method references use)

File: docs/src/main/java/GettingStarted.java (hunk 2 above)
Comment (saurabhagas) [reply docs/src/main/java/GettingStarted.java:59]: Dynamic config changes are available in certain kits only (e.g. ehcache 3.9-SNAPSHOT, tc-ee 10.7.0-SNAPSHOT). Using these kits automatically translates to `Distribution107Controller`. Using a global `EHCACHE_OS_VERSION` didn't seem correct, because the added tests need to use a different ehcache version

File: docs/src/main/java/GettingStarted.java (hunk 3 above)
Comment (saurabhagas) [reply docs/src/main/java/GettingStarted.java:110]: Dynamic config changes are available in certain kits only (e.g. ehcache 3.9-SNAPSHOT, tc-ee 10.7.0-SNAPSHOT). Using these kits automatically translates to `Distribution107Controller`. Using a global `EHCACHE_OS_VERSION` didn't seem correct, because the added tests need to use a different ehcache version

File: docs/src/main/asciidoc/getting-started.adoc (hunk 4 above)
Comment (saurabhagas) [reply docs/src/main/asciidoc/getting-started.adoc:70]: Okay, will take your comment and rephrase as needed.

Comment (saurabhagas) [reply docs/src/main/asciidoc/getting-started.adoc:73]: Done

File: client-internal/src/main/java/org/terracotta/angela/client/Tsa.java (hunk 1 above)
Comment (aurbroszniowski) [reply client-internal/src/main/java/org/terracotta/angela/client/Tsa.java:765]: if that is linked to the method name, it would be helpful for debugging to also log the parameters that were passed

File: docs/src/main/java/GettingStarted.java (hunk 2 above)
Comment (aurbroszniowski) [reply docs/src/main/java/GettingStarted.java:59]: you can update the EHCACHE_OS_VERSION field to use 3.9-SNAPSHOT

Comment (saurabhagas) [reply docs/src/main/java/GettingStarted.java:59]: The old angela API cannot be used for dynamic-config kits. The old APIs need to use the non-dynamic-config kits.

File: client-internal/src/main/java/org/terracotta/angela/client/Tsa.java (hunk 1 above)
Comment (saurabhagas) [reply client-internal/src/main/java/org/terracotta/angela/client/Tsa.java:765]: That is happening already inside the config tool invocation :-)

File: docs/src/main/java/GettingStarted.java (hunk 2 above)
Comment (aurbroszniowski) [reply docs/src/main/java/GettingStarted.java:59]: I'm wondering  what would be the best:
have 2 different tests files for documentation, one for <3.9, the other for >=3.9
or delete the <3.9 test in the documentation
or maybe keep one java file and have two variables EHCACHE_OS_VERSION_OLDER_LINE and EHCACHE_OS_VERSION

but the fact that yo can't use one variable now and needs to hard code the 3.9 version shows that something is missing

Comment (saurabhagas) [reply docs/src/main/java/GettingStarted.java]: Introduced a new variable called `EHCACHE_VERSION`. Ambiguous on purpose (didn't like OLDER/NEWER distinction)

**Comment by saurabhagas** [comment]
@aurbroszniowski removed the tests from this PR for now. I need to debug the failures locally. Feel free to merge this PR until then.
